master/buildbot/reporters/stash.py

```diff
@@ -82,3 +90,52 @@ def send(self, build):
                 content = yield response.content()
                 log.error("{code}: Unable to send Stash status: {content}",
                           code=response.code, content=content)
+
+
+class StashPRCommentPush(http.HttpStatusPushBase):
+    name = "StashPRCommentPush"
+
+    @defer.inlineCallbacks
+    def reconfigService(self, base_url, user, password, text=None, verbose=False, **kwargs):
+        yield http.HttpStatusPushBase.reconfigService(self, wantProperties=True,
+                                                      **kwargs)
+        self.text = text or Interpolate('Builder: %(prop:buildername)s Status: %(prop:statustext)s')
+        self.verbose = verbose
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, base_url, auth=(user, password))
+
+    @defer.inlineCallbacks
+    def send(self, build):
+        if build['complete'] \
+           and build['properties'].has_key("pullrequesturl"):
+                yield self.sendPullRequestComment(build)
+
+    @defer.inlineCallbacks
+    def sendPullRequestComment(self, build):
+        props = Properties.fromDict(build['properties'])
+        pr_url = props.getProperty("pullrequesturl")
+        match = re.search("^(http|https)://([^/]+)/(.+)$", pr_url)
```

Comment:
this is meant to get the path as shown later:
`        path = match.group(3)`

maybe I should add something like "# this matches {protocol}://{host}/{path} in order to extract the path"

The second group ([^/]+) is done that way in order to avoid backtracking and improve the performance